File: TicTacToe/ttt.js
const readline = require("readline");
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

let board = Array.from({length:3},()=>Array(3).fill(" "));
let player = "X";

function printBoard() {
  board.forEach((r,i)=>{
    console.log(" " + r.join(" | "));
    if(i<2) console.log("---+---+---");
  });
}

function win(p) {
  for(let i=0;i<3;i++)
    if(board[i].every(c=>c===p) || board.map(r=>r[i]).every(c=>c===p))
      return true;
  return [0,1,2].every(i=>board[i][i]===p) ||
         [0,1,2].every(i=>board[i][2-i]===p);
}

function draw() {
  return board.flat().every(c=>c!==" ");
}

function ask() {
  printBoard();
  rl.question(`\nPlayer ${player} (row col or q): `, ans=>{
    if(ans==="q") return rl.close();
    let [r,c]=ans.split(" ").map(Number);
    if(!r||r<1||r>3||c<1||c>3||board[r-1][c-1]!==" ") {
      console.log("Invalid move!");
      return ask();
    }
    board[r-1][c-1]=player;
    if(win(player)) { printBoard(); console.log(`${player} wins!`); return rl.close(); }
    if(draw()) { printBoard(); console.log("Draw!"); return rl.close(); }
    player = player==="X"?"O":"X";
    ask();
  });
}

ask();

File: TicTacToe/ttt.py
board = [[" "]*3 for _ in range(3)]

def print_board():
    for i in range(3):
        print(" | ".join(board[i]))
        if i < 2:
            print("-"*9)

def win(p):
    return any(
        all(board[r][c] == p for c in range(3)) or
        all(board[c][r] == p for c in range(3))
        for r in range(3)
    ) or all(board[i][i] == p for i in range(3)) or \
       all(board[i][2-i] == p for i in range(3))

def draw():
    return all(cell != " " for row in board for cell in row)

player = "X"

while True:
    print_board()
    try:
        move = input(f"\nPlayer {player} (row col) or q: ")
        if move.lower() == "q":
            break
        r, c = map(int, move.split())
        if not (1 <= r <= 3 and 1 <= c <= 3):
            print("Out of range!")
            continue
        r -= 1; c -= 1
        if board[r][c] != " ":
            print("Cell occupied!")
            continue
        board[r][c] = player
        if win(player):
            print_board()
            print(f"\nPlayer {player} wins!")
            break
        if draw():
            print_board()
            print("\nDraw!")
            break
        player = "O" if player == "X" else "X"
    except:
        print("Invalid input!")
